src/NCronJob/Execution/JobExecutor.cs

```diff
@@ -80,13 +77,25 @@ public async Task RunJob(JobRun run, CancellationToken stoppingToken)
     private IJob ResolveJob(IServiceProvider scopedServiceProvider, JobDefinition definition)
     {
         if (definition.Type == typeof(DynamicJobFactory))
-            return dynamicJobFactoryRegistry.GetJobInstance(scopedServiceProvider, definition);
+            return jobRegistry.GetDynamicJobInstance(scopedServiceProvider, definition);
 
         var job = scopedServiceProvider.GetService(definition.Type);
         if (job != null)
             return (IJob)job;
 
         LogUnregisteredJob(definition.Type);
+
+        // This relies on reflection and goes in the way of AOT compilation.
+        // (cf. https://github.com/NCronJob-Dev/NCronJob/issues/54)
+        //
+        // Maybe add a .UseNCronJob() extension method that would harvest the JobRegistry
+        // and ensure every described type has been previously registered.
+        // Otherwise, throw an InvalidOperationException ("The following types haven't
```

Comment:
We could link https://github.com/NCronJob-Dev/NCronJob/issues/37 here - or just handle it in a complete different topic.

I try to keep the "TODO"s at a minimum.